syntax = "proto3";

option java_multiple_files = true;
option java_package = "org.athenian.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";


package helloworld;

// The Greeter service definition.
service Greeter {
    rpc SayHello (HelloRequest) returns (HelloReply) {
    }

    rpc SayHelloWithManyRequests (stream HelloRequest) returns (HelloReply) {
    }

    rpc SayHelloWithManyReplies (HelloRequest) returns (stream HelloReply) {
    }

    rpc SayHelloWithManyRequestsAndReplies (stream HelloRequest) returns (stream HelloReply) {
    }
}

// The request message containing the user's name.
message HelloRequest {
    string name = 1;
}

// The response message containing the greetings
message HelloReply {
    string message = 1;
}
